src/NServiceBus.Core/Recoverability/Recoverability.cs

```diff
@@ -0,0 +1,170 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using Config;
+    using ConsistencyGuarantees;
+    using Features;
+    using Settings;
+    using Transports;
+
+    class Recoverability : Feature
+    {
+        public Recoverability()
+        {
+            EnableByDefault();
+            DependsOnOptionally<DelayedDeliveryFeature>();
+
+            Prerequisite(context => !context.Settings.GetOrDefault<bool>("Endpoint.SendOnly"),
+                "Message recoverability is only relevant for endpoints receiving messages.");
+            Defaults(settings =>
+            {
+                settings.SetDefault(ImmedidateRetriesEnabled, true);
+                settings.SetDefault(DelayedRetriesEnabled, true);
+                settings.SetDefault(FailureInfoStorageCacheSizeKey, 1000);
+            });
+        }
+        protected internal override void Setup(FeatureConfigurationContext context)
+        {
+            var errorQueue = context.Settings.ErrorQueueAddress();
+            context.Settings.Get<QueueBindings>().BindSending(errorQueue);
+
+            var transportTransactionMode = context.Settings.GetRequiredTransactionModeForReceives();
+            var failureInfoStorage = new FailureInfoStorage(context.Settings.Get<int>(FailureInfoStorageCacheSizeKey));
+            var localAddress = context.Settings.LocalAddress();
+
+            var incomingStepSequence = context.Pipeline.Register("MoveFaultsToErrorQueue", b => new MoveFaultsToErrorQueueBehavior(
```

Comment:
why explicitly defining a step id? (and creating the required registration api?)